From af165b4fc87135c324715bb3b19c330461d2f8e6 Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@freescale.com>
Date: Mon, 5 Jun 2017 11:17:05 +0800
Subject: [PATCH 1855/5242] MLK-15003-3: ARM64: dts: add one more cell in edma
 note

commit  f8d04626fa6e75b57ef72303dc21009e945ff0e3 from
https://source.codeaurora.org/external/imx/linux-imx.git

Add one more cell in edma note, which is the property of
local/remote access. The default access is local.

Signed-off-by: Shengjiu Wang <shengjiu.wang@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi  |   30 ++++++++++++------------
 arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi |    8 +++----
 2 files changed, 19 insertions(+), 19 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi
index e864a20..b4653c8 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi
@@ -1131,8 +1131,8 @@
 		assigned-clock-rates = <80000000>;
 		power-domains = <&pd_dma_lpuart1>;
 		dma-names = "tx","rx";
-		dmas = <&edma0 15 0 0>,
-			<&edma0 14 0 1>;
+		dmas = <&edma0 15 0 0 0>,
+			<&edma0 14 0 1 0>;
 		status = "disabled";
 	};
 
@@ -1148,8 +1148,8 @@
 		assigned-clock-rates = <80000000>;
 		power-domains = <&pd_dma_lpuart2>;
 		dma-names = "tx","rx";
-		dmas = <&edma0 17 0 0>,
-			<&edma0 16 0 1>;
+		dmas = <&edma0 17 0 0 0>,
+			<&edma0 16 0 1 0>;
 		status = "disabled";
 	};
 
@@ -1165,8 +1165,8 @@
 		assigned-clock-rates = <80000000>;
 		power-domains = <&pd_dma_lpuart3>;
 		dma-names = "tx","rx";
-		dmas = <&edma0 19 0 0>,
-			<&edma0 18 0 1>;
+		dmas = <&edma0 19 0 0 0>,
+			<&edma0 18 0 1 0>;
 		status = "disabled";
 	};
 
@@ -1182,8 +1182,8 @@
 		assigned-clock-rates = <80000000>;
 		power-domains = <&pd_dma_lpuart4>;
 		dma-names = "tx","rx";
-		dmas = <&edma0 21 0 0>,
-			<&edma0 20 0 1>;
+		dmas = <&edma0 21 0 0 0>,
+			<&edma0 20 0 1 0>;
 		status = "disabled";
 	};
 
@@ -1199,7 +1199,7 @@
 		      <0x0 0x5a330000 0x0 0x10000>, /* channel19 UART3 tx */
 		      <0x0 0x5a340000 0x0 0x10000>, /* channel20 UART4 rx */
 		      <0x0 0x5a350000 0x0 0x10000>; /* channel21 UART4 tx */
-		#dma-cells = <3>;
+		#dma-cells = <4>;
 		dma-channels = <10>;
 		interrupts = <GIC_SPI 434 IRQ_TYPE_LEVEL_HIGH>,
 			     <GIC_SPI 435 IRQ_TYPE_LEVEL_HIGH>,
@@ -1233,7 +1233,7 @@
 			<0x0 0x59290000 0x0 0x10000>, /* spdif0 tx */
 			<0x0 0x592c0000 0x0 0x10000>, /* sai0 rx */
 			<0x0 0x592d0000 0x0 0x10000>; /* sai0 tx */
-		#dma-cells = <3>;
+		#dma-cells = <4>;
 		shared-interrupt;
 		dma-channels = <12>;
 		interrupts = <GIC_SPI 374 IRQ_TYPE_LEVEL_HIGH>, /* asrc0 */
@@ -1536,7 +1536,7 @@
 			<&clk IMX8QM_CLK_DUMMY>,
 			<&clk IMX8QM_CLK_DUMMY>;
 		clock-names = "core", "extal", "fsys", "spba";
-		dmas = <&edma2 6 0 1>, <&edma2 7 0 0>;
+		dmas = <&edma2 6 0 1 0>, <&edma2 7 0 0 0>;
 		dma-names = "rx", "tx";
 		power-domains = <&pd_esai0>;
 		status = "disabled";
@@ -1562,7 +1562,7 @@
 			      "rxtx3", "rxtx4",
 			      "rxtx5", "rxtx6",
 			      "rxtx7", "spba";
-		dmas = <&edma2 8 0 1>, <&edma2 9 0 0>;
+		dmas = <&edma2 8 0 1 0>, <&edma2 9 0 0 0>;
 		dma-names = "rx", "tx";
 		power-domains = <&pd_spdif0>;
 		status = "disabled";
@@ -1579,7 +1579,7 @@
 			<&clk IMX8QM_CLK_DUMMY>;
 		clock-names = "bus", "mclk0", "mclk1", "mclk2", "mclk3";
 		dma-names = "rx", "tx";
-		dmas = <&edma2 12 0 1>, <&edma2 13 0 0>;
+		dmas = <&edma2 12 0 1 0>, <&edma2 13 0 0 0>;
 		status = "disabled";
 		power-domains = <&pd_sai0>;
 	};
@@ -1614,8 +1614,8 @@
 			"asrck_8", "asrck_9", "asrck_a", "asrck_b",
 			"asrck_c", "asrck_d", "asrck_e", "asrck_f",
 			"spba";
-		dmas = <&edma2 0 0 0>, <&edma2 1 0 0>, <&edma2 2 0 0>,
-			<&edma2 3 0 1>, <&edma2 4 0 1>, <&edma2 5 0 1>;
+		dmas = <&edma2 0 0 0 0>, <&edma2 1 0 0 0>, <&edma2 2 0 0 0>,
+			<&edma2 3 0 1 0>, <&edma2 4 0 1 0>, <&edma2 5 0 1 0>;
 		dma-names = "rxa", "rxb", "rxc",
 				"txa", "txb", "txc";
 		fsl,asrc-rate  = <8000>;
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
index 29c66fc..8b99522 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
@@ -826,7 +826,7 @@
 			<0x0 0x59290000 0x0 0x10000>, /* spdif0 tx */
 			<0x0 0x592c0000 0x0 0x10000>, /* sai0 rx */
 			<0x0 0x592d0000 0x0 0x10000>; /* sai0 tx */
-		#dma-cells = <3>;
+		#dma-cells = <4>;
 		shared-interrupt;
 		dma-channels = <12>;
 		interrupts = <GIC_SPI 374 IRQ_TYPE_LEVEL_HIGH>, /* asrc0 */
@@ -868,7 +868,7 @@
 			<&clk IMX8QXP_CLK_DUMMY>;
 		clock-names = "bus", "mclk0", "mclk1", "mclk2", "mclk3";
 		dma-names = "rx", "tx";
-		dmas = <&edma2 12 0 1>, <&edma2 13 0 0>;
+		dmas = <&edma2 12 0 1 0>, <&edma2 13 0 0 0>;
 		status = "disabled";
 		power-domains = <&pd_sai0>;
 	};
@@ -903,8 +903,8 @@
 			"asrck_8", "asrck_9", "asrck_a", "asrck_b",
 			"asrck_c", "asrck_d", "asrck_e", "asrck_f",
 			"spba";
-		dmas = <&edma2 0 0 0>, <&edma2 1 0 0>, <&edma2 2 0 0>,
-			<&edma2 3 0 1>, <&edma2 4 0 1>, <&edma2 5 0 1>;
+		dmas = <&edma2 0 0 0 0>, <&edma2 1 0 0 0>, <&edma2 2 0 0 0>,
+			<&edma2 3 0 1 0>, <&edma2 4 0 1 0>, <&edma2 5 0 1 0>;
 		dma-names = "rxa", "rxb", "rxc",
 				"txa", "txb", "txc";
 		fsl,asrc-rate  = <8000>;
-- 
1.7.9.5

